### ISSUE TITLE
Error: Handshake failed: no matching key exchange algorithm

### ISSUE THREAD

**Issue by stamatimorellas** [maitrungduc1410/node-scp-async#18]
I am getting an error when attempting to download a file locally via scp: `Error: Handshake failed: no matching key exchange algorithm`

I have tried using other libraries for `scp` and `ssh` within Node.js and I have seen this error a few times with different packages. My ssh key is stored in a file `~/.ssh/id_rsa` and I use a variable `SSH_PRIVATE_KEY_PATH` for reference when looking at my code below. 

To put things in perspective, I am trying to first access a remote machine to create and populate a file, and then I am trying to get that file to my local machine. I am using the `simple-ssh` package to access the remote host and run the commands necessary to generate my file; for this connection, I only supply *hostname*, *username*, and *password* and NO private key, and it seems to work fine. 

Here is what my code looks like when I try to get the file to my local device using `node-scp`:

```js
const { Client } = require('node-scp');

const USER_HOME = require('os').homedir;
const SSH_PRIVATE_KEY_PATH = USER_HOME + '/.ssh/id_rsa';

function getSettingsFromHost(ip, uname, pw, file) {

    const remoteFile = '/tmp/' + file;
    const localFile = appRoot + '/tmp/' + file;

    Client({
        host: ip,
        port: 22,
        username: uname,
        password: pw,
        privateKey: fs.readFileSync(SSH_PRIVATE_KEY_PATH)
    }).then(client => {
        client.downloadFile(remoteFile, localFile)
            .then(response => {
                console.log(Colors.GREEN, "[SUCCESS] Got file from host " + ip);
                client.close();
            }).catch(err => console.log(Colors.RED, "[ERROR] " + err + ". Terminating..."))
    }).catch(error => console.log(Colors.RED, "[ERROR] " + error));
}

```

Am I missing something?

**Comment by lopezhansel** [comment]
@stamatimorellas, I recently started having a very similar issue after a server update. I am able to connect using `scp` through the terminal, but not with node and `node-scp`. None of my credentials have changed remotely nor locally and I am also using a privateKey too. Oddly, toggling `tryKeyboard`  on allows me to connect properly. Give it a try and hopefully this can help others. 
```js
Client({
  host: ip,
  port: 22,
  username: uname,
  password: pw,
  privateKey: fs.readFileSync(SSH_PRIVATE_KEY_PATH),
  tryKeyboard: true
})
```

**Comment by stamatimorellas** [comment]
@lopezhansel Thank you for this!! I also tried the same thing you did (scp through terminal, programatically) using 

```js
spawn("scp", [uname + "@" + ip + ":" + remoteFilePath, localDirPath]);
```

which works, but hopefully adding the `tryKeyboard` config value will allow me to use `node-scp` library as I originally intended.

Cheers!

**Comment by maitrungduc1410** [comment]
Hi all, new version of `node-scp` is available (0.0.16), with full support for every connection options from `ssh2`, check README for new updates. Thank you
